diff --git a/src/Authenticators/TokenAuthenticator.php b/src/Authenticators/TokenAuthenticator.php
index fea397e..41c0b8c 100644
--- a/src/Authenticators/TokenAuthenticator.php
+++ b/src/Authenticators/TokenAuthenticator.php
@@ -166,7 +166,7 @@ class TokenAuthenticator implements Authenticator
                     $member->{$tokenDBColumn} = $tokenData['token'];
                     $member->{$expireDBColumn} = $tokenData['expire'];
                     $member->write();
-                    $member->login();
+                    Injector::inst()->get(IdentityStore::class)->logIn($member, false, $request);
                 }
             }
 
@@ -201,7 +201,7 @@ class TokenAuthenticator implements Authenticator
 
         if ($member) {
             //logout
-            $member->logout();
+            Injector::inst()->get(IdentityStore::class)->logOut($request);
 
             if ($this->tokenConfig['owner'] === Member::class) {
                 //generate expired token
diff --git a/src/RESTfulAPI.php b/src/RESTfulAPI.php
index 4e57526..e7a4363 100644
--- a/src/RESTfulAPI.php
+++ b/src/RESTfulAPI.php
@@ -515,7 +515,7 @@ class RESTfulAPI extends Controller
     private static function api_access_config_check($className, $httpMethod = 'GET')
     {
         $access = false;
-        $api_access = singleton($className)->stat('api_access');
+        $api_access = singleton($className)->config()->get('api_access');
 
         if (is_string($api_access)) {
             $api_access = explode(',', strtoupper($api_access));
diff --git a/src/Serializers/DefaultSerializer.php b/src/Serializers/DefaultSerializer.php
index 502768b..b11a538 100644
--- a/src/Serializers/DefaultSerializer.php
+++ b/src/Serializers/DefaultSerializer.php
@@ -152,7 +152,7 @@ class DefaultSerializer implements Serializer
         $apiFields = (array) Config::inst()->get($class, 'api_fields');
 
         //$many_many_extraFields = $dataObject->many_many_extraFields();
-        $many_many_extraFields = $dataObject->stat('many_many_extraFields');
+        $many_many_extraFields = $dataObject->config()->get('many_many_extraFields');
 
         // setup ID (not included in $db!!)
         $serializedColumnName = $this->serializeColumnName('ID');
